android/jni/tun2socks/tun2socks.c

```diff
@@ -1153,10 +1164,23 @@ int process_arguments (void)
             return 0;
         }
     }
-    // resolve local DNS server address
-    if (options.dns_server_address) {
-        if (!BAddr_Parse2(&dns_server_address, options.dns_server_address, NULL, 0, 0)) {
-            BLog(BLOG_ERROR, "local dns server addr: BAddr_Parse2 failed");
+
+    if (options.transparent_dns) {
+        if (!options.dns_resolver_address) {
+            BLog(BLOG_ERROR,
+                 "transparentDNS requires a DNS resolver address");
+            return 0;
```

Comment:
It looks like in some places (below) where this was returning 0, this changeset replaces that with `goto fail` (which just does `return 0`), but not here. Would it be clearer to instead have constants like `ERRCODE = 0` and `SUCCESSCODE = 1`, and replace all the existing `return 0`, `return 1`, and `goto fail` statements with `return ERRCODE` or `return SUCCESSCODE`?

(I also noticed this is the opposite of process / `main` function return codes, where 0 is success and nonzero is failure. Worth flipping?)